code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -246,6 +246,75 @@
 /datum/aiTask/succeedable/repair/on_reset()
 	has_started = 0
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+// DEPOSIT GOAL
+// targets: ghost tealprints in the same flock
+// precondition: 10 resources
+/datum/aiTask/sequence/goalbased/deposit
+	name = "depositing"
+	weight = 4
+
+/datum/aiTask/sequence/goalbased/deposit/New(parentHolder, transTask)
+	..(parentHolder, transTask)
+	add_task(holder.get_instance(/datum/aiTask/succeedable/deposit, list(holder)))
+
+/datum/aiTask/sequence/goalbased/deposit/precondition()
+	. = 0
+	var/mob/living/critter/flock/drone/F = holder.owner
+	if(F?.can_afford(10))
+		. = 1
+
+/datum/aiTask/sequence/goalbased/deposit/on_reset()
+	var/mob/living/critter/flock/drone/F = holder.owner
+	if(F)
+		F.active_hand = 2 // nanite spray
+		sleep(0.1 SECONDS)
+		F.a_intent = INTENT_HELP
+		F.hud?.update_intent()
+		sleep(0.1 SECONDS)
+		F.hud?.update_hands() // for observers
```

Comment:
what's with these sleeps

Reply:
I just copied the Repair task, so if anyone knows it's probably @moonlol

Reply:
im not really sure what its meant to accomplish, maybe without the sleeps it doesn't work show up right for observers?

Reply:
I'd probably at least test if it works with the sleeps removed. If it does remove them, if it doesn't keep them for now and it can be refactored later.